vortex-array/src/array/constant/compute.rs

```diff
@@ -59,6 +68,45 @@ impl SearchSortedFn for ConstantArray {
     }
 }
 
+impl AndFn for ConstantArray {
+    fn and(&self, array: &Array) -> VortexResult<Array> {
+        constant_array_bool_impl(self, array, |(l, r)| l & r)
+    }
+}
+
+impl OrFn for ConstantArray {
+    fn or(&self, array: &Array) -> VortexResult<Array> {
+        constant_array_bool_impl(self, array, |(l, r)| l | r)
+    }
+}
+
+fn constant_array_bool_impl(
+    constant_array: &ConstantArray,
+    other: &Array,
+    bool_op: impl Fn((bool, bool)) -> bool,
+) -> VortexResult<Array> {
+    if constant_array.dtype().is_boolean()
+        && other.dtype().is_boolean()
+        && constant_array.len() == other.len()
+    {
+        if let Ok(array) = ConstantArray::try_from(other.clone()) {
+            let lhs = constant_array.scalar().value().as_bool()?;
+            let rhs = array.scalar().value().as_bool()?;
+
+            let scalar = match lhs.zip(rhs).map(bool_op) {
+                Some(b) => Scalar::bool(b, Nullability::Nullable),
+                None => Scalar::null(constant_array.dtype().as_nullable()),
+            };
+
+            Ok(ConstantArray::new(scalar, constant_array.len()).into_array())
+        } else {
+            AndFn::and(&constant_array.clone().into_bool()?, other)
+        }
+    } else {
+        vortex_bail!("Boolean operations aren't supported on arrays of different lengths")
```

Comment:
This is the style point we were discussing 😅 

I'd rather invert the if statement and bail at the start of the function. That way the bail is kept close to the logic that decides it, and the rest of the function doesn't need to be indented by another 4 spaces.

Reply:
Changed the function to fail early, I'm willing to concede that here it looks better :)